### PR TITLE
Don't call WebRequest::getLimitOffset in 1.35

### PR DESCRIPTION
**Pull request by DannyS712**
See https://phabricator.wikimedia.org/T245779

### CODE REVIEW

File: includes/specials/SpecialProperties.php [hunk 1]
```diff
@@ -66,7 +66,13 @@ protected function getGroupName() {
...
 
 	private function getLimitOffset() {
-		return $this->getRequest()->getLimitOffset();
+		$request = $this->getRequest();
+		if ( method_exists( $request, 'getLimitOffsetForUser' ) ) {
+			// MW 1.35+
+			return $request->getLimitOffsetForUser( $this->getUser() );
+		} else {
+			return $request->getLimitOffset();
+		}
```
Comment (JeroenDeDauw) [inline includes/specials/SpecialProperties.php]: You can write this without the else part:

```php
		if ( method_exists( $request, 'getLimitOffsetForUser' ) ) {
			// MW 1.35+
			return $request->getLimitOffsetForUser( $this->getUser() );
		}

		return $request->getLimitOffset();
```

Comment (DannyS712) [reply includes/specials/SpecialProperties.php]: True, but I was using the same code block that I was using elsewhere where it wasn't returning, but rather setting a variable, so I kept the else block

Comment (JeroenDeDauw) [reply includes/specials/SpecialProperties.php]: That is part of the reason why using a guard clause (the form in my variation) is nice.

Comment (JeroenDeDauw) [reply includes/specials/SpecialProperties.php]: https://www.entropywins.wtf/blog/2019/01/14/readable-functions-guard-clause/

**Comment by kghbln** [comment]
@DannyS712 Thanks a lot for the pull!

**Pull request merged**